crates/re_ui/src/lib.rs

```diff
@@ -515,6 +515,70 @@ impl ReUi {
         response
     }
 
+    /// Popup similar to [`egui::popup_below_widget`] but suitable for use with
+    /// [`crate::list_item::ListItem`].
+    pub fn list_item_popup<R>(
+        ui: &egui::Ui,
+        popup_id: egui::Id,
+        widget_response: &egui::Response,
+        vertical_offset: f32,
+        add_contents: impl FnOnce(&mut egui::Ui) -> R,
+    ) -> Option<R> {
+        if ui.memory(|mem| mem.is_popup_open(popup_id)) {
+            let pos = widget_response.rect.left_bottom() + egui::vec2(0.0, vertical_offset);
+            let pivot = Align2::LEFT_TOP;
+
+            let inner = egui::Area::new(popup_id)
+                .order(egui::Order::Foreground)
+                .constrain(true)
+                .fixed_pos(pos)
+                .pivot(pivot)
+                .show(ui.ctx(), |ui| {
+                    let frame = egui::Frame {
+                        fill: ui.visuals().panel_fill,
+                        ..Default::default()
+                    };
+                    let frame_margin = frame.total_margin();
+                    frame
+                        .show(ui, |ui| {
+                            ui.with_layout(
+                                egui::Layout::top_down_justified(egui::Align::LEFT),
+                                |ui| {
+                                    ui.set_width(
+                                        widget_response.rect.width() - frame_margin.sum().x,
+                                    );
+
+                                    ui.set_clip_rect(ui.cursor());
+
+                                    egui::ScrollArea::vertical()
+                                        .show(ui, |ui| {
+                                            egui::Frame {
+                                                //TODO(ab): use design token
+                                                inner_margin: egui::Margin::symmetric(8.0, 0.0),
+                                                ..Default::default()
+                                            }
+                                            .show(ui, add_contents)
+                                            .inner
+                                        })
+                                        .inner
+                                },
+                            )
+                            .inner
+                        })
+                        .inner
+                })
+                .inner;
```

Comment:
@emilk is there a way to avoid this repeated `.inner` thing? feels like something is wrong :/

Reply:
Given that the `inner` type is not already an `Option` that fix feels slightly worse than the actual issue 🤷🏻‍♂️

Reply:
The return-type is an `Option` though, so it works.